src/main/java/org/gridsuite/directory/server/DirectoryService.java

```diff
@@ -52,11 +57,19 @@ private static ElementAttributes fromEntity(DirectoryElementEntity entity) {
     }
 
     public Flux<ElementAttributes> listDirectoryContent(String directoryUuid) {
-        return Flux.fromStream(directoryElementRepository.findByParentId(UUID.fromString(directoryUuid)).stream().map(DirectoryService::fromEntity));
+        return Flux.fromStream(directoryContentStream(directoryUuid));
+    }
+
+    private Stream<ElementAttributes> directoryContentStream(String directoryUuid) {
+        return directoryElementRepository.findByParentId(UUID.fromString(directoryUuid)).stream().map(DirectoryService::toElementAttributes);
+    }
+
+    public Mono<DirectoryAttributes> getDirectoryInfos(String directoryUuid) {
+        return Mono.fromCallable(() -> directoryElementRepository.findById(UUID.fromString(directoryUuid)).map(DirectoryService::toDirectoryAttributes).orElse(null));
     }
 
     public Flux<ElementAttributes> getRootDirectories() {
-        return Flux.fromStream(directoryElementRepository.findByParentId(null).stream().map(DirectoryService::fromEntity));
+        return Flux.fromStream(directoryElementRepository.findByParentId(null).stream().map(DirectoryService::toElementAttributes));
```

Comment:
Why not return a Flux<DirectoryAttributes> instead of Flux<ElementAttributes> since the returned values are directories ?